packages/starlight/utils/plugins.ts

```diff
@@ -16,8 +16,24 @@ export async function runPlugins(
 	// Validate the user-provided configuration.
 	let userConfig = starlightUserConfig;
 
+	const starlightSchema =
+		context.config.output === 'static'
+			? StarlightConfigSchema.and(
+					z.object({
+						prerender: z
+							.literal(true, {
+								errorMap: () => ({
+									message:
+										'astro:Disabling prerendering is not supported when using the `static` output mode.',
+								}),
+							})
+							.optional(),
+					})
+			  )
+			: StarlightConfigSchema;
```

Comment:
I assume this change is due to me commenting in my previous review that the 2 previous checks should be moved to the schema. I think the change is great for the other check altho I now realize that I missed in my previous review the fact that this other check also use the `context`.

I'm wondering if for this one specific check, we could only validate once, when all the plugins have ran with a simple `if` like you had before (basically just moving the previous code near the end of the function)? This would avoid this schema "dynamic" change and also the changes in `packages/starlight/utils/error-map.ts` with the `astro:` prefix which I'm not a huge fan of.

Do you have any thoughts?

Reply:
I think validating only once after the plugins would be great. The tradeoff is that we lose the report of which plugin caused the problem in case one of them disables prerender. I think that is fine tho.